### ISSUE TITLE
New Loss Format

### ISSUE THREAD

**Issue by hfaghihi15** [HLR/DomiKnowS#301]
Hi all, 

I think we need to implement a better loss function setting where each learner also receives the loss function attached to it. But apart from that we also need to be able to attach masking to the losses. 

I think the best way to do so is to define mask property on the LabelReaderSensor. Let me know what you think?

The system should contain, masking, weight integration, different and custom loss functions, applying different loss functions during training, pertaining, pipelines, and so on.
